serialization.md

```diff
@@ -84,27 +80,16 @@ Address/Hash Helpers
 ```
 
 - `#hashSignedTx` Takes transaction data. Returns the hash of the rlp-encoded transaction with R S and V.
-- `#hashUnsignedTx` Returns the hash of the rlp-encoded transaction without R S or V.
 
 ```k
-    syntax String ::= #hashSignedTx   ( Int , Int , Int , Account , Int , ByteArray , Int , ByteArray , ByteArray ) [function]
-                    | #hashUnsignedTx ( Int , Int , Int , Int , Account , Int , ByteArray , Int , JSONs )           [function]
-                    | #hashUnsignedTx ( Int , Int , Int , Account , Int , ByteArray, Int )                          [function]
- // --------------------------------------------------------------------------------------------------------------------------
+    syntax String ::= #hashSignedTx( Int , Int , Int , Account , Int , ByteArray , Int , ByteArray , ByteArray ) [function]
+                    | #hashTxData  ( TxData )                                                                    [function]
+ // -----------------------------------------------------------------------------------------------------------------------
     rule #hashSignedTx(TN, TP, TG, TT, TV, TD, TW, TR, TS)
       => Keccak256( #rlpEncode([ TN, TP, TG, #addrBytes(TT), TV, TD, TW, TR, TS ]) )
 
-    // Hashing for legacy transactions
-    rule #hashUnsignedTx(0, TN, TP, TG, TT, TV, TD, _CID, _TA)
-      => Keccak256( #rlpEncode([ TN, TP, TG, #addrBytes(TT), TV, TD ]) )
-
-    // Hashing for EIP-2930: Optional access lists
-    rule #hashUnsignedTx(1, TN, TP, TG, TT, TV, TD, CID, [TA])
-      => Keccak256( "\x01" +String #rlpEncode([ CID, TN, TP, TG, #addrBytes(TT), TV, TD, [TA] ]) )
-
-    // Hashing for EIP-155: Simple replay attack protection
-    rule #hashUnsignedTx(TN, TP, TG, TT, TV, TD, CID)
-      => Keccak256( #rlpEncode([ TN, TP, TG, #addrBytes(TT), TV, TD, CID, "", "" ]) )
+    rule #hashTxData( TXDATA )                           => Keccak256( #rlpEncodeTxData(TXDATA) ) [owise]
+    rule #hashTxData( AccessListTxData(...) #as TXDATA ) => Keccak256( "\x01" +String #rlpEncodeTxData(TXDATA) )
```

Comment:
@ehildenb It looks like the java backend doesn't support my use of `#as` here. Are there any solutions that would be nicer than writing out the entire production?